server/src/main/java/org/elasticsearch/index/mapper/SortedSetDocValuesSyntheticFieldLoaderLayer.java

```diff
@@ -133,6 +148,87 @@ public void write(XContentBuilder b) throws IOException {
         }
     }
 
+    private class OffsetDocValuesLoader implements DocValuesLoader, DocValuesFieldValues {
+        private final BinaryDocValues oDv;
+        private final SortedSetDocValues dv;
+        private final ByteArrayStreamInput scratch = new ByteArrayStreamInput();
+
+        private boolean hasValue;
+        private int[] offsetToOrd;
+
+        OffsetDocValuesLoader(SortedSetDocValues dv, BinaryDocValues oDv) {
+            this.dv = dv;
+            this.oDv = oDv;
+        }
+
+        @Override
+        public boolean advanceToDoc(int docId) throws IOException {
+            hasValue = dv.advanceExact(docId);
+            if (hasValue) {
+                if (oDv.advanceExact(docId)) {
+                    var encodedValue = oDv.binaryValue();
+                    scratch.reset(encodedValue.bytes, encodedValue.offset, encodedValue.length);
+                    offsetToOrd = scratch.readVIntArray();
+                } else {
+                    offsetToOrd = null;
+                }
+                return true;
+            } else {
+                offsetToOrd = null;
+                return false;
+            }
+        }
+
+        @Override
+        public int count() {
+            if (hasValue) {
+                if (offsetToOrd != null) {
+                    // HACK: trick CompositeSyntheticFieldLoader to serialize this layer as array.
+                    // (if offsetToOrd is not null, then at index time an array was always specified even if there is just one value)
+                    return offsetToOrd.length + 1;
+                } else {
+                    return dv.docValueCount();
+                }
+            } else {
+                return 0;
+            }
+        }
+
+        @Override
+        public void write(XContentBuilder b) throws IOException {
+            if (hasValue == false) {
+                return;
+            }
+            if (offsetToOrd != null) {
+                long[] ords = new long[dv.docValueCount()];
+                for (int i = 0; i < dv.docValueCount(); i++) {
+                    ords[i] = dv.nextOrd();
+                }
+
+                logger.info("ords=" + Arrays.toString(ords));
```

Comment:
I guess you are using `info` just for debugging while in draft.

Reply:
Yes, this was for debugging purposes. This will be removed.